src/Controller/BlogPostController.php

```diff
@@ -0,0 +1,207 @@
+<?php
+declare(strict_types=1);
+
+namespace App\Controller;
+
+use App\DTO\BlogPostEnquiry;
+use App\DTO\BlogPostListEnquiry;
+use App\Entity\BlogPost;
+use App\Entity\BlogUser;
+use App\Repository\BlogUserRepository;
+use App\Service\BlogPostService;
+use App\Service\Serializer\DTOSerializer;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\Exception\NotSupported;
+use Doctrine\ORM\OptimisticLockException;
+use Doctrine\ORM\ORMException;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+use App\Repository\BlogPostRepository;
+
+class BlogPostController extends AbstractController
+{
+
+    /**
+     * @Route("/blogposts/{userID?}", name="blog-posts", methods={"GET"})
+     * @throws NotSupported
+     * @throws \JsonException
```

Comment:
Semantics, but when I call `/blogposts` it returns list regardless of user, however when `/blogposts/` is used, the request's `getContent()` returns an empty string instead, even though it's still capturing the`userID=null`. Why is that?

Reply:
Might be framework specific.
Actual assumption, `/blogposts/` should still be able to retrieve body content

Reply:
Upon using query parameters, `/blogposts/` and `/blogposts` act as intended (not anymore dependent on `getContent()` as `GET` should only be used to request data anyway